testcode/lib/gamate/nachtm.c

```diff
@@ -0,0 +1,1147 @@
+    int                 SoftEnv;
+    const unsigned*     Data;           /* Pointer to data */
+    int                 DecVal;
+} VoiceCtrl;
+
+/* Control structs for all three voices */
+static VoiceCtrl V1 = {
+    0x01, 0x11, 0, 0, 0, Voice1, 0,
+};
+static VoiceCtrl V2 = {
+    0x02, 0x41, 0, 0, 0, Voice2, 0,
+};
+static VoiceCtrl V3 = {
+    0x04, 0x11, 0, 0, 0, Voice3, 0,
+};
+
+/* Pointers to the structs for easy reference */
+static VoiceCtrl* V [3] = {
+    &V1, &V2, &V3
+};
+
+/* Screen dimensions */
+static unsigned char XSize, YSize;
+
+/* Variable that contains the time of the next clock tick to play a note */
+static unsigned char NextClock;
+
+/* Start- and runtime */
+static clock_t StartTime;
+
+/* Number of ticks for each tone */
+#define TICKS_PER_TONE  10
+
+/* Done flag. Contains one bit for each voice. Will contain 0x07 if all
+** voices have finished playing.
+*/
+static unsigned char Done;
+
+#  define outb(addr,val)        (*(addr)) = (val)
+#  define outw(addr,val)        (*(addr)) = (val)
+
+unsigned char       I, N, Offs;
+unsigned char       Tone;
+unsigned char       Octave;
+unsigned            Val;
+
+/*****************************************************************************/
+/*                                   Code                                    */
+/*****************************************************************************/
+
+static void MakeTeeLine (unsigned char Y)
+/* Make a divider line */
+{
+    cputcxy (0, Y, CH_LTEE);
+    chline (XSize - 2);
+    cputc (CH_RTEE);
+}
+
+static void MakeNiceScreen (void)
+/* Make a nice screen */
+{
+    typedef struct {
+        unsigned char   Y;
+        char*           Msg;
+    } TextDesc;
+    static TextDesc Text [] = {
+              /*12345678901234567890*/
+        {   1, "Wolfgang A. Mozart"  },
+        {   2, "Eine kleine"         },
+        {   3, "Nachtmusik"          },
+        {   4, "(KV 525)"            },
+        {   6, "Ported to Gamate"    },
+        {   7, "Groepaz / Hitmen"},
+    };
+
+    register const TextDesc* T;
+    unsigned char I;
+    unsigned char X;
+
+    /* Clear the screen hide the cursor, set colors */
+    textcolor (COLOR_BLACK);
+    bgcolor (COLOR_WHITE);
+    clrscr ();
+    cursor (0);
+
+    /* Top line */
+    cputcxy (0, 0, CH_ULCORNER);
+    chline (XSize - 2);
+    cputc (CH_URCORNER);
+
+    /* Left line */
+    cvlinexy (0, 1, YSize - 2);
+
+    /* Bottom line */
+    cputc (CH_LLCORNER);
+    chline (XSize - 2);
+    cputc (CH_LRCORNER);
+
+    /* Right line */
+    cvlinexy (XSize - 1, 1, YSize - 2);
+
+    /* Several divider lines */
+    MakeTeeLine (5);
+    MakeTeeLine (8);
+
+    /* Write something into the frame */
+    for (I = 0, T = Text; I < sizeof (Text) / sizeof (Text [0]); ++I) {
+        X = (XSize - strlen (T->Msg)) / 2;
+        cputsxy (X, T->Y, T->Msg);
+        ++T;
+    }
+}
+
+void UpdateScreen(void)
+{
+    register VoiceCtrl* VC;
+    /* Play all three voices */
+    for (I
```

Comment:
I'm certainly willing to be less style nit-picking for test code but I still think one source file should have at least a consistent style - and this isn't the case in this file regarding blanks before actual parameter lists.